swarm-test/src/lib.rs

```diff
@@ -0,0 +1,226 @@
+use async_trait::async_trait;
+use futures::future::Either;
+use futures::{AsyncRead, AsyncWrite, StreamExt};
+use libp2p::core::upgrade::Version;
+use libp2p::identity::Keypair;
+use libp2p::noise::NoiseAuthenticated;
+use libp2p::swarm::{
+    AddressScore, ConnectionHandler, IntoConnectionHandler, NetworkBehaviour, SwarmEvent,
+};
+use libp2p::yamux::YamuxConfig;
+use libp2p::{Multiaddr, PeerId, Swarm, Transport};
+use std::fmt::Debug;
+use std::time::Duration;
+
+type THandler<TBehaviour> = <TBehaviour as NetworkBehaviour>::ConnectionHandler;
+type THandlerErr<TBehaviour> =
+    <<THandler<TBehaviour> as IntoConnectionHandler>::Handler as ConnectionHandler>::Error;
+
+/// An extension trait for [`Swarm`] that makes it easier to set up a network of [`Swarm`]s for tests.
+#[async_trait]
+pub trait SwarmExt {
+    type NB: NetworkBehaviour;
+
+    /// Create a new [`Swarm`] with an ephemeral identity.
+    fn new_ephemeral<T>(
+        transport: T,
+        behaviour_fn: impl FnOnce(PeerId, Keypair) -> Self::NB,
+    ) -> Self
+    where
+        T: Transport + Send + 'static + Unpin,
+        T::Dial: Send + 'static,
+        T::ListenerUpgrade: Send + 'static,
+        T::Error: Send + Sync,
+        T::Output: AsyncRead + AsyncWrite + Send + 'static + Unpin,
+        Self: Sized;
+
+    /// Establishes a connection to the given [`Swarm`], polling both of them until the connection is established.
+    async fn block_on_connection<T>(&mut self, other: &mut Swarm<T>)
```

Comment:
```suggestion
    async fn connect<T>(&mut self, other: &mut Swarm<T>)
```

- `block_on` reminds me of `futures::executor::block_on` which this doesn't use.
- `block_on_connection` does not imply actually triggering the dial

What do you think of the new name?

Reply:
Yeah sounds good and it is shorter too :)